megameklab/src/megameklab/ui/mek/BMLAMFuelView.java

```diff
@@ -0,0 +1,155 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megameklab.ui.mek;
+
+import megamek.common.*;
+import megamek.common.util.EncodeControl;
+import megameklab.ui.EntitySource;
+import megameklab.ui.listeners.MekBuildListener;
+import megameklab.ui.util.IView;
+import megameklab.util.UnitUtil;
+
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import java.util.List;
+import java.util.ResourceBundle;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Collectors;
+
+/**
+ * This view block allows entering additional fuel tanks for LandAirMeks
+ *
+ * @author Simon (Juliez)
+ */
+public class BMLAMFuelView extends IView implements ChangeListener {
+
+    private static final EquipmentType FUEL_TANK = MiscType.get(EquipmentTypeLookup.LAM_FUEL_TANK);
+
+    private final List<MekBuildListener> listeners = new CopyOnWriteArrayList<>();
+    public void addListener(MekBuildListener l) {
+        listeners.add(l);
+    }
+    public void removeListener(MekBuildListener l) {
+        listeners.remove(l);
+    }
+
+    private final ResourceBundle resourceMap = ResourceBundle.getBundle("megameklab.resources.Views", new EncodeControl());
+    private final SpinnerNumberModel fuelTonsSpnModel = new SpinnerNumberModel(0, 0, 84, 1);
+    private final JSpinner fuelTonsSpn = new JSpinner(fuelTonsSpnModel);
+    private final JLabel totalFuelPointsLabel = new JLabel();
+
+    public BMLAMFuelView(EntitySource entitySource) {
+        super(entitySource);
+        initUI();
+    }
+
+    private void initUI() {
+        setLayout(new BoxLayout(this, BoxLayout.PAGE_AXIS));
+
+        fuelTonsSpn.setToolTipText(resourceMap.getString("BMLAMFuelView.lblFuelPoints.tooltip"));
+        fuelTonsSpn.addChangeListener(this);
+
+        Box fuelRangePanel = Box.createHorizontalBox();
+        fuelRangePanel.add(Box.createHorizontalStrut(10));
+        fuelRangePanel.add(new JLabel(resourceMap.getString("BMLAMFuelView.spnFuel.text")));
+        fuelRangePanel.add(Box.createHorizontalStrut(10));
+        fuelRangePanel.add(fuelTonsSpn);
+        fuelRangePanel.add(new JLabel(" tons"));
+        fuelRangePanel.add(Box.createHorizontalStrut(10));
+        fuelRangePanel.setBorder(new EmptyBorder(0, 0, 10, 0));
+        JPanel totalPointsPanel = new JPanel();
+        totalPointsPanel.add(totalFuelPointsLabel);
+
+        add(fuelRangePanel);
+        add(totalPointsPanel);
+        add(Box.createVerticalStrut(5));
+    }
+
+    public void setFromEntity(Mech mek) {
+        fuelTonsSpnModel.setValue(fuelTanks(mek));
+        updateFuelPointsLabel();
+    }
+
+    /** Returns the current number of LAM Fuel Tanks on the Mek, including unallocated. Returns 0 for non-LAM. */
```

Comment:
Can we please convert this into proper Javadoc formatting? It'll be a solid improvement to have it as such once I finish working on the Javadoc backend setup.